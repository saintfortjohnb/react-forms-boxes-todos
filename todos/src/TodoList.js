import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList() {
  const [todos, setTodos] = useState([]);

  const addTodo = (newTodo) => {
    setTodos([...todos, newTodo]);
  };

  const removeTodo = (id) => {
    setTodos(todos.filter(todo => todo.id !== id));
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      {todos.map(todo => (
        <Todo key={todo.id} id={todo.id} task={todo.task} removeTodo={removeTodo} />
      ))}
    </div>
  );
}

export default TodoList;
