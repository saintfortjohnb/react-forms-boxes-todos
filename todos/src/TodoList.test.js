import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

// Smoke Test
it('TodoList renders without crashing', () => {
  render(<TodoList />);
});

// Snapshot Test
it('TodoList matches snapshot', () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it('adds a new todo', () => {
  const { getByRole, getByText } = render(<TodoList />);

  const input = getByRole('textbox');
  const button = getByText('Add Todo');

  fireEvent.change(input, { target: { value: 'New Task' } });
  fireEvent.click(button);
  
  expect(getByText('New Task')).toBeInTheDocument();
});

// Logic Test for Deleting Todo
it('deletes a todo', () => {
  const { getByRole, getByText } = render(<TodoList />);
  
  const input = getByRole('textbox');
  const addButton = getByText('Add Todo');
  
  fireEvent.change(input, { target: { value: 'Task to delete' } });
  fireEvent.click(addButton);

  const deleteButton = getByText('X');
  fireEvent.click(deleteButton);
  
  expect(() => getByText('Task to delete')).toThrow();
});
