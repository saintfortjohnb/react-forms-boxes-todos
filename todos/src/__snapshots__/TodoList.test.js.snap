// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
